source/common/upstream/load_balancer_impl.cc

```diff
@@ -593,14 +593,31 @@ HostConstSharedPtr EdfLoadBalancerBase::chooseHostOnce(LoadBalancerContext* cont
 
 HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector& hosts_to_use,
                                                                 const HostsSource&) {
-  // This is just basic unweighted P2C.
-  const HostSharedPtr host1 = hosts_to_use[random_.random() % hosts_to_use.size()];
-  const HostSharedPtr host2 = hosts_to_use[random_.random() % hosts_to_use.size()];
-  if (host1->stats().rq_active_.value() < host2->stats().rq_active_.value()) {
-    return host1;
-  } else {
-    return host2;
+  // A user is not allowed to specify less than two choices.
+  uint32_t choice_count = 2;
+  if (least_request_config_.has_value()) {
+    choice_count = least_request_config_.value().choice_count().value();
+  }
+
```

Comment:
Probably want to `ASSERT(choice_count >= 2)`

Reply:
I don't think this one made it in, but otherwise LGTM.

Reply:
Yikes, forgot to push. Thanks.